docs/shared/recipes/plugins/project-graph-plugins.md

```diff
@@ -80,6 +80,57 @@ If you create targets for a project within a plugin's code, the Nx migration gen
 
 {% /callout %}
 
+#### Example (extending projects / adding targets)
+
+A plugin can also be written to modify existing projects. Most of Nx's first party are written to add a target to a given project based on the configuration files present for that project. The below example shows how a plugin could add a target to a project based on the presence of a `tsconfig.json` file.
+
+```typescript {% fileName="/my-plugin/index.ts" %}
+export const createNodes: CreateNodes = [
+  '**/tsconfig.json',
+  (fileName: string, opts, context: CreateNodesContext) => {
+    const root = dirname(fileName);
+
+    const isProject = existsSync(join(root, 'project.json'));
+    if (!isProject) {
+      return {};
+    }
+
+    return {
+      projects: {
+        [root]: {
+          targets: {
+            build: {
+              command: `tsc -p ${fileName}`,
+            },
+          },
+        },
+      },
+    };
+  },
+];
+```
+
+By checking for the presence of a `project.json` file, the plugin can ensure that the project it is modifying is an existing Nx project.
+
+When extending an existing project, its important to consider how Nx will merge the returned project configurations. In general, plugins are ran in the order they are listed in `nx.json`, with Nx's built-in plugins being ran last. Plugins overwrite information that was identified by plugins that ran before them if a merge is not possible.
```

Comment:
```suggestion
When extending an existing project, its important to consider how Nx will merge the returned project configurations. In general, plugins are run in the order they are listed in `nx.json`, with Nx's built-in plugins are run last. Plugins overwrite information that was identified by plugins that run before them if a merge is not possible.
```

Reply:
The plugin execution order is the opposite of what I would expect.  Since users have no control over Nx's built-in plugins, I would expect them to be executed first so that users could add plugins that modify the default behavior.  This would also open the door to third-party plugins being able to make decisions based on what the built-in plugins have already done.

I know this is a bigger technical design decision and outside the scope of this PR, but it doesn't make sense to me.

Reply:
When I'm talking about "built in plugins" what I'm really referencing are:
- Target Defaults
- `project.json`
- `package.json` / package manager workspaces

These come last / override other plugins because they are explicit configuration formats for Nx itself. You wouldn't want an inferred config to override the explicit config, so they come last.

Does that make sense?

It is a bit rough that we don't know what all projects Nx has identified at this point, but by the same token, we don't want to add functionality here that can't be replicated by non-core plugins. If we moved the built-in plugins outside of the regular plugin flow and executed them first to build up some context object (to contain the projects identified) that would mean that the package-manager workspaces plugin's identified projects would be visible, but projects identified by nx-dotnet for example wouldn't be. Plugins couldn't rely on that list of projects to be complete necessarily, and it would be harder to eventually move some stuff over to @nx/js.